#! /bin/sh /usr/share/dpatch/dpatch-run
## cairobug12284-gitd62f8861689d8b9a9a837043fb78813f0407abd4.dpatch by Dave Beckett <dajobe@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Early detection of a zero sized bitmap
## DP: Cairo bug 12284

From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 5 Sep 2007 07:07:41 +0000 (+0100)
Subject: [cairo-ft-font] Early detection of a zero sized bitmap.
X-Git-Tag: 1.5.2
X-Git-Url: http://gitweb.freedesktop.org/?p=cairo;a=commitdiff;h=d62f8861689d8b9a9a837043fb78813f0407abd4

[cairo-ft-font] Early detection of a zero sized bitmap.

Under rare circumstances we may need to extract a surface that
represents a bitmap with width==0 and rows==0. Detect this case at the
start and simply return a zero-sized surface.

Fixes https://bugs.freedesktop.org/show_bug.cgi?id=12284.
---

--- a/src/cairo-ft-font.c
+++ b/src/cairo-ft-font.c
@@ -735,6 +735,12 @@ _get_bitmap_surface (FT_Bitmap		     *bi
     width = bitmap->width;
     height = bitmap->rows;
 
+    if (width == 0 || height == 0) {
+	*surface = (cairo_image_surface_t *)
+	    cairo_image_surface_create_for_data (NULL, format, 0, 0, 0);
+	return (*surface)->base.status;
+    }
+
     switch (bitmap->pixel_mode) {
     case FT_PIXEL_MODE_MONO:
 	stride = (((width + 31) & ~31) >> 3);
